### PR TITLE
Added getRepoData to parent class metrics

### PR DESCRIPTION
**Pull request by RushilShaw**
Almost every child function was running a separate version of the getRepoData function, this has been simplified by adding the function to the parent class Metrics, and adding the class variables owner and repo. All child classes have been adjusted.

### CODE REVIEW

File: src/Metrics.ts [hunk 1]
```diff
@@ -19,9 +19,22 @@ export abstract class Metrics {
...
 
     constructor(url: string) {
         this.url = url;
+        const { owner, repo } = this.getRepoData(this.url);
+        this.owner = owner;
+        this.repo = repo;
+        
+    }
+
+    private getRepoData(url: string): { owner: string; repo: string } {
```
Comment (RushilShaw) [inline src/Metrics.ts:33]: a note: this is no longer a async function

**Comment by RushilShaw** [comment]
As, I have refactored the codebase, I am humbly requesting that everyone review how I have altered each persons' respective function.

**Review by galaxybomb23 — COMMENTED** [review]
looks good, submit a SS of it passing tests pls

**Comment by RushilShaw** [comment]
Screenshot 2024-09-10 at 4 33 26 PM

**Review by iBoot32 — APPROVED** [review]
let's get this moneyyyyyy

**Comment by iBoot32** [comment]
all tests have failed, however i simply do not care. merge this.

**Pull request merged**